backend/typescript/services/implementations/adminService.ts

```diff
@@ -0,0 +1,134 @@
+import prisma from "../../prisma";
+import type {
+  IAdminService,
+  NotificationDTO,
+} from "../interfaces/adminService";
+// import type ResidentDTO from "../interfaces/FILEOFDTO";
+// import type IStaffService from "../interfaces/FILEOFDTO";
+import logger from "../../utilities/logger";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import { IResidentService } from "../interfaces/residentService";
+import ResidentService from "./residentService";
+
+const residentService: IResidentService = new ResidentService();
+const Logger = logger(__filename);
+
+class AdminService implements IAdminService {
+  staffId: number;
+
+  constructor(staffId: number) {
+    this.staffId = staffId;
+  }
+
+  async getAllNotifications(): Promise<NotificationDTO[]> {
+    try {
+      const notifications = await prisma.notification.findMany({
+        include: { residents: true },
+      });
+      if (!notifications) throw new Error(`No residents found.`);
+      return notifications;
+    } catch (error) {
+      Logger.error(
+        `Failed to get all active Residents. Readon = ${getErrorMessage(
+          error,
+        )}`,
+      );
+      throw error;
+    }
+  }
+
+  async getNotificationById(id: number): Promise<NotificationDTO> {
+    try {
+      const notification = await prisma.notification.findUnique({
+        where: {
+          id: Number(id),
```

Comment:
I think it might be better to use just id here, but in the adminResolver file parse the id input as a Number (aka change line 16 to adminService.getNotificationById(Number(id)) This way the value is parsed as a Number earlier and so it has to be parsed less (in the general case).